src/taskProvider.ts

```diff
@@ -0,0 +1,76 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as serverUtils from './omnisharp/utils';
+import * as vscode from './vscodeAdapter';
+
+import { OmniSharpServer } from './omnisharp/server';
+import { TaskRevealKind } from 'vscode';
+
+let taskProvider: vscode.Disposable | undefined;
+
+export function activate(_context: vscode.ExtensionContext, server: OmniSharpServer): void {
+	if (!vscode.workspace.workspaceFolders) {
+		return;
+	}
+
+	taskProvider = vscode.workspace.registerTaskProvider('dotnet', {
```

Comment:
Where did the 'dotnet' constant come from?

Reply:
It's defined in package.json. I chose this constant fairly arbitrarily. Would we prefer `c#`, `csharp`, `.net`?